terraform/alerts.tf

```diff
@@ -0,0 +1,66 @@
+resource "aws_sns_topic" "alarm" {
+  name = "${local.prefix}-alarms-${local.short_uuid}"
+  tags = local.tags
+}
+
+resource "aws_cloudwatch_metric_alarm" "lambda_errors" {
+  for_each = {
+    publish = aws_lambda_function.async_publish.function_name,
+    webhook = aws_lambda_function.webhook_github.function_name,
+    get     = aws_lambda_function.modules_get.function_name,
+    list    = aws_lambda_function.modules_list.function_name,
+    builds  = aws_lambda_function.builds_get.function_name,
+  }
+
+  alarm_name          = "${local.prefix}-lambda-errors-alarm-${each.key}-${local.short_uuid}"
+  comparison_operator = "GreaterThanThreshold"
+  evaluation_periods  = 1
+  period              = 300
+  datapoints_to_alarm = 1
+  statistic           = "Sum"
+  metric_name         = "Errors"
+  namespace           = "AWS/Lambda"
+  dimensions = {
+    "FunctionName" = each.value
+  }
+  threshold          = 10
+  treat_missing_data = "missing"
+  alarm_actions      = [aws_sns_topic.alarm.arn]
+  tags               = local.tags
+}
+
+resource "aws_cloudwatch_metric_alarm" "stuck_builds" {
+  alarm_name          = "${local.prefix}-build-queue-old-messages-${local.short_uuid}"
+  comparison_operator = "GreaterThanThreshold"
+  evaluation_periods  = 1
+  period              = 300
+  datapoints_to_alarm = 1
+  statistic           = "Maximum"
+  metric_name         = "ApproximateAgeOfOldestMessage"
+  namespace           = "AWS/SQS"
+  dimensions = {
+    "QueueName" = aws_sqs_queue.build_queue.name
+  }
+  threshold          = 900 // 15 minutes
+  treat_missing_data = "missing"
+  alarm_actions      = [aws_sns_topic.alarm.arn]
+  tags               = local.tags
+}
+
+resource "aws_cloudwatch_metric_alarm" "dlq_messages" {
+  alarm_name          = "${local.prefix}-total-build-dlq-${local.short_uuid}"
+  comparison_operator = "GreaterThanThreshold"
+  evaluation_periods  = 1
+  period              = 300
+  datapoints_to_alarm = 1
+  statistic           = "Average"
+  metric_name         = "ApproximateNumberOfMessagesVisible"
+  namespace           = "AWS/SQS"
+  dimensions = {
+    "QueueName" = aws_sqs_queue.build_dlq.name
+  }
+  threshold          = 25
```

Comment:
Does this mean 25 messages need to be in the DLQ for the alarm to go off?

Reply:
Yup (plus or minus some, SQS can't give an exact number)

Reply:
:+1: - I don't expect this alert to ever trigger anyway because the max age alert will likely kick in first.

Reply:
yup, it's really just to cover our bases